### ISSUE TITLE
Error when training with my own data 

### ISSUE THREAD

**Issue by jiayily** [vislearn/LessMore#6]
Hello, I have tried your trained model on 7scenes' fire dataset. The result is pretty good. Now I want to train a model with my own data, i.e  training without depth images. But when I run the command “../../train_repro -oscript train_obj_nomodel.lua -omodel obj_model_fcn_init_nomodel.net”, I get this error:

**
Round 65490 of 300000.
Predicting object coordinates.
CNN prediction took 0.0254037s.
Calculating gradients wrt projection error.
Projection Loss: 86.9206
Gradient statistics:
Max gradient: 0.773935
Avg gradient: 0.0245269
Med gradient: 0

Done in 0.0167138s.
Done in 0.0176077s.

Round 65491 of 300000.
Predicting object coordinates.
CNN prediction took 0.0273097s.
Calculating gradients wrt projection error.
Projection Loss: 84.1978
Gradient statistics:
Max gradient: 0.72726
Avg gradient: 0.0263427
Med gradient: 0

Done in 0.0163486s.
PANIC: unprotected error in call to Lua API (stack overflow)
**
Do you have some advice? Thank you

**Comment by ebrach** [comment]
Hi,

the error is very unspecific, unfortunately. So I cannot say immediately whats wrong. Can you maybe also paste the console output when you start training (up to the first iteration)? Also, did the first training stage (train_obj) succeed? (In the sense that the training loss decreased. For how many iterations did you run train_obj?

Kind regards,
Eric

**Comment by jiayily** [comment]
The training iterations is the same as in your code. The output of train_obj, i.e. training_loss_train_obj_nomodel.lua_.txt is very strange:
0 0
1 0
2 0
3 0
4 0
5 0
6 0
7 0
8 0
9 0
10 0
...
This means the loss is 0 all the time?

**Comment by jiayily** [comment]
Sorry， I forgot to make soft links.

**Comment by FangGet** [comment]
@ebrach I also trained with my own dataset(rgb, depth, poses).
1. when do train_obj(step 1), the loss decreased in general while vibrate a lot.
2. after 200K iterations, I started to do train_repro(step 2), but **the loss never converge or decrease**.
Do you have any advice for this? or Is it convenient for you to provide a training log for your 7Scenes or other dataset?
Thank you.

**Comment by ebrach** [comment]
Hi, 
1) If the loss vibrates a lot, there seem to be a few training images which are very hard. This could be normal for your data or maybe hint to an error in the data setup (maybe worth investigating).
2) What loss do you observe for train_repro? 
Kind regards,
Eric

**Comment by FangGet** [comment]
Hi,
   Sorry for late reply. for the dataset quality, I checked it and it's OK though less quality than 7Scenes/chess for pose drift, I convert it just like chess's format.
for tran_repro loss, it also vibrate a lot, between 40 and 80, and never decrease, with seldom huge value like 200 appear. I checked you code and fond nothing abnormal. I decide to continue next step and see the final behavior.

**Comment by ebrach** [comment]
Well, if the loss does not decrease, that's usually not a good sign. Let me know how it went. If it does not work out, you could try to overfit on a single image (or a few images). If the loss still would not decrease, there is something wrong.
